vendor/open-cluster-management.io/api/cluster/v1beta1/types_placement.go

```diff
@@ -168,19 +168,30 @@ type ClusterPredicate struct {
 type ClusterSelector struct {
 	// LabelSelector represents a selector of ManagedClusters by label
 	// +optional
-	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty"`
+	LabelSelector ClusterLabelSelector `json:"labelSelector,omitempty"`
 
 	// ClaimSelector represents a selector of ManagedClusters by clusterClaims in status
 	// +optional
 	ClaimSelector ClusterClaimSelector `json:"claimSelector,omitempty"`
 }
 
+type ClusterLabelSelector struct {
+	// +optional
+	metav1.LabelSelector `json:",inline"`
+
+	// +optional
+	RegexSelector map[string]string `json:"regexSelector,omitempty"`
```

Comment:
I would consider a CEL based selector, which could be more flexible.
e.g. could reference a score value